torch_np/_ndarray.py

```diff
@@ -410,14 +411,32 @@ def clip(self, min, max, out=None):
     )
 
     ### indexing ###
-    def __getitem__(self, *args, **kwds):
-        t_args = _helpers.ndarrays_to_tensors(*args)
-        return ndarray._from_tensor_and_base(
-            self._tensor.__getitem__(*t_args, **kwds), self
-        )
+    @staticmethod
+    def _upcast_int_indices(index):
+        if isinstance(index, torch.Tensor):
+            if index.dtype in [torch.int8, torch.int16, torch.int32]:
```

Comment:
Adding `torch.uint8` here should fix a few issues below.

Also, minor nit, but in general, prefer tuples over lists.

Reply:
> Also, minor nit, but in general, prefer tuples over lists.

IMO I prefer lists for any homogenous sequence... or rather, [because Raymond Hettinger said so](https://twitter.com/raymondh/status/324664257004322817?lang=en-GB) :sweat_smile:

Reply:
It's a rather tiny optimisation, but well, I like constant things, and, on top of that, small lists are constructed off tuples
```python
>>> dis("[1,2,3]")
  1           0 BUILD_LIST               0
              2 LOAD_CONST               0 ((1, 2, 3))
              4 LIST_EXTEND              1
              6 RETURN_VALUE
>>> dis("(1,2,3)")
  1           0 LOAD_CONST               0 ((1, 2, 3))
              2 RETURN_VALUE
```